smartsim/_core/launcher/slurm/slurmCommands.py

```diff
@@ -120,3 +110,18 @@ def _find_slurm_command(cmd: str) -> str:
         raise LauncherError(
             f"Slurm Launcher could not find path of {cmd} command"
         ) from e
+
+
+def _execute_slurm_cmd(
```

Comment:
It seems like all the WLMs would benefit from this type of check and two other options might be preferable:

- add `def ensure_preconditions()` to the launcher contract and execute that when launching an experiment instead of getting down to an individual cmd. This lets every launcher add their own check for preconditions and gives a quicker, clearer indicator of what failed (bonus: it can be executed in the preview to show a user that their experiment won't complete)
- generalize `_find_slurm_command` into `_find_required_command(..., owner: str)` in shell.py. Pass in "Slurm"/"PBS" and avoid every WLM copying the same 30 lines

Reply:
As per our private conversation, we will create a ticket for this and won't implement it as part of this ticket. 🚀